src/simple_controller_impl/src/main.rs.reference

```diff
@@ -0,0 +1,112 @@
+// Nightly clippy (0.1.64) considers Drop a side effect, see https://github.com/rust-lang/rust-clippy/issues/9608
```

Comment:
I assume this is the normally written controller code we are trying to match?

Reply:
Yes. This reference impl is slightly more complex (for example it checks the return error of create to decide whether to return error or ok)